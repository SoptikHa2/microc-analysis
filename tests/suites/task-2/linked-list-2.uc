list_append(list, x) {
    var node;
    node = alloc {v: x, prev: list, next: null};
    if (list == null) {
    } else {
        (*list).next = node;
    }

    return node;
}