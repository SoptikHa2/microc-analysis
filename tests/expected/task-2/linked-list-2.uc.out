---
test: type suites/task-2/linked-list-2.uc
---
⟦list_append(list[1:13],x[1:19]){...}:[1:1]⟧ = (↑{v:t6,prev:t7,next:↑µt8.{v:t5,prev:↑{v:t6,prev:t7,next:↑t8},next:↑t9}},t5) -> ↑µt8.{v:t5,prev:↑{v:t6,prev:t7,next:↑t8},next:↑t9}
⟦list[1:13]⟧ = ↑{v:t6,prev:t7,next:↑µt8.{v:t5,prev:↑{v:t6,prev:t7,next:↑t8},next:↑t9}}
⟦x[1:19]⟧ = t5
⟦node[2:9]⟧ = ↑µt8.{v:t5,prev:↑{v:t6,prev:t7,next:↑t8},next:↑t9}

exit: 0
